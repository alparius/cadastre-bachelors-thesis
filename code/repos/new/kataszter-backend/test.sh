#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
