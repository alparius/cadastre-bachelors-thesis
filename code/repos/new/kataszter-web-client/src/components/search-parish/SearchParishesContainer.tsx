import { LatLngExpression } from "leaflet";
import React from "react";
import { FormattedMessage as FM, useIntl } from "react-intl";
import { Map, Marker, Popup, TileLayer, Tooltip } from "react-leaflet";
import MarkerClusterGroup from "react-leaflet-markercluster";
import { Card, Container, Header, Icon, Pagination } from "semantic-ui-react";

import { NavLink } from "react-router-dom";
import IParishMin from "../../data/parish/ParishMin";
import ItalicFM from "../../util/styledMessages/ItalicFM";
import CardPlaceholder from "../CardPlaceholder";
import ErrorMsg from "../ErrorMsg";
import ParishCard from "./ParishCard";

const PAGESIZE: number = 10;

interface ISearchParishesContainerProps {
    parishes: IParishMin[] | undefined;
    loading: IParishMin[] | boolean;
    showError: boolean;
    isListView: boolean;
    currentPage: number;
    setCurrentPage(newPage: number): void;
}

const SearchParishesContainer: React.FC<ISearchParishesContainerProps> = (props: ISearchParishesContainerProps) => {
    const { parishes, loading, showError, isListView, currentPage, setCurrentPage } = props;
    const { formatMessage: fm } = useIntl();

    const handlePageChange = (_: any, { activePage }: any) => setCurrentPage(activePage);

    if (showError) {
        return <ErrorMsg errorMessage={"error.connection"} />;
    } else if (loading) {
        return <CardPlaceholder />;
    } else if (!parishes || parishes.length === 0) {
        return (
            <>
                <CardPlaceholder />
                <Header as="h2">
                    <FM id="search.noResult" />
                </Header>
            </>
        );
    } else if (!isListView) {
        const noMap =
            parishes.length -
            parishes.filter((parish: IParishMin) => parish.coordinates.lat !== undefined && parish.coordinates.lng !== undefined).length;

        return (
            <>
                <Header as="h2">
                    {parishes.length} <FM id="search.parishResult" />
                    {noMap > 0 && <FM id="search.parishResultNoCoordinates" values={{ noMap }} />}
                </Header>
                <Header as="h5" style={{ marginTop: "-1em" }}>
                    <FM id="search.switchView" />
                </Header>
                <Map className="markercluster-map" center={{ lat: 46.769, lng: 23.589 } as LatLngExpression} zoom={7}>
                    <TileLayer
                        attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
                        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
                    />
                    <MarkerClusterGroup maxClusterRadius={15} showCoverageOnHover={false} removeOutsideVisibleBounds>
                        {parishes.map(
                            (parish: IParishMin) =>
                                parish.coordinates.lat &&
                                parish.coordinates.lng && (
                                    <Marker position={[parish.coordinates.lat, parish.coordinates.lng]} key={parish.ID}>
                                        <Popup>
                                            <Header as="h2" className="hoverLink">
                                                <NavLink to={`/parishes/profile/${parish.ID}`}>{parish.name}</NavLink>
                                            </Header>
                                            <Container>
                                                <FM id="label.numberOfPriests" />:{" "}
                                                {parish.numberOfPriests ? parish.numberOfPriests : <ItalicFM id="general.noData" />}
                                                <br />
                                                <FM id="label.parish.cityName" />:{" "}
                                                {parish.cityName ? parish.cityName : <ItalicFM id="general.noData" />}
                                                <span style={{ float: "right" }}>
                                                    {parish.verified ? (
                                                        <>
                                                            {fm({ id: "profile.verified.isVerifiedPopupShort" })}{" "}
                                                            <Icon name="check circle outline" fitted color="green" />
                                                        </>
                                                    ) : (
                                                        <>
                                                            {fm({ id: "profile.verified.notVerifiedPopupShort" })}{" "}
                                                            <Icon name="question circle outline" fitted color="grey" />
                                                        </>
                                                    )}
                                                </span>
                                            </Container>
                                        </Popup>
                                        <Tooltip>{parish.name}</Tooltip>
                                    </Marker>
                                )
                        )}
                    </MarkerClusterGroup>
                </Map>
            </>
        );
    } else {
        return (
            <>
                {currentPage === 1 && (
                    <Header as="h2">
                        {parishes.length} <FM id="search.parishResult" />
                    </Header>
                )}
                <Card.Group stackable doubling itemsPerRow={2}>
                    {parishes.slice((currentPage - 1) * PAGESIZE, currentPage * PAGESIZE).map((parish: IParishMin) => (
                        <ParishCard key={parish.ID} parish={parish} />
                    ))}
                </Card.Group>
                <Container textAlign="center" style={{ marginTop: "3em" }}>
                    <Pagination activePage={currentPage} totalPages={Math.ceil(parishes.length / PAGESIZE)} onPageChange={handlePageChange} />
                </Container>
            </>
        );
    }
};

export default SearchParishesContainer;
