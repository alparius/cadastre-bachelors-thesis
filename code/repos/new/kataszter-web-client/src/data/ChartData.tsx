interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
