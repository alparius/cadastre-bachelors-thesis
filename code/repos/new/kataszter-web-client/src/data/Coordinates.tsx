interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
