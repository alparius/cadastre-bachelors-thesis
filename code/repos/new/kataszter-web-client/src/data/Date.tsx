interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
