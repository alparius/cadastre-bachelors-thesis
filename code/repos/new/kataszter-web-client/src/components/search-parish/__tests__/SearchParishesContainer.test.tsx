/*===================
  SearchParishesContainer.unit.test.tsx
 ===================*/
import React from "react";

import IParishMin from "../../../data/parish/ParishMin";
import { NewTestParishMin } from "../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter } from "../../../util/test/test-wrapper";
import SearchParishesContainer from "../SearchParishesContainer";

const handlers = {
    setCurrentPage: jest.fn()
};

const parishes: IParishMin[] = [NewTestParishMin];

describe("SearchParishesContainer element", () => {
    afterEach(() => {
        handlers.setCurrentPage.mockClear();
    });

    it("should render a card group when the parishes array is not empty", () => {
        const container = createComponentWithIntlAndRouter(
            <SearchParishesContainer parishes={parishes} isListView loading={false} showError={false} currentPage={1} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a placeholder card and a message when the parishes array is empty", () => {
        const emptyParishes: IParishMin[] = [];
        const container = createComponentWithIntlAndRouter(
            <SearchParishesContainer parishes={emptyParishes} isListView loading={false} showError={false} currentPage={1} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render an error message when showError is true", () => {
        const container = createComponentWithIntlAndRouter(
            <SearchParishesContainer parishes={parishes} isListView loading={false} showError currentPage={1} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render an error message when showError is true, for map view", () => {
        const container = createComponentWithIntlAndRouter(
            <SearchParishesContainer parishes={parishes} isListView={false} loading={false} showError currentPage={1} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render an error message when showError and loading are both true", () => {
        const container = createComponentWithIntlAndRouter(
            <SearchParishesContainer parishes={parishes} isListView loading showError currentPage={1} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render an error message when showError and loading are both true, for map view", () => {
        const container = createComponentWithIntlAndRouter(
            <SearchParishesContainer parishes={parishes} isListView={false} loading showError currentPage={1} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a placeholder card when only loading is true", () => {
        const container = createComponentWithIntlAndRouter(
            <SearchParishesContainer parishes={parishes} isListView loading showError={false} currentPage={1} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a placeholder card when only loading is true, for map view", () => {
        const container = createComponentWithIntlAndRouter(
            <SearchParishesContainer parishes={parishes} isListView={false} loading showError={false} currentPage={1} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });
});
