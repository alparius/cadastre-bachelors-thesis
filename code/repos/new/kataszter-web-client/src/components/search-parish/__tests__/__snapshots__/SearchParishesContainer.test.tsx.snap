// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchParishesContainer element should render a card group when the parishes array is not empty 1`] = `
Array [
  <h2
    className="ui header"
  >
    1
     
    Parishes found matching your search
  </h2>,
  <div
    className="ui doubling stackable two cards"
  >
    <a
      aria-current={null}
      className="ui fluid card"
      href="/parishes/profile/1"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          <h2
            className="ui header"
          >
            name
            <span
              style={
                Object {
                  "float": "right",
                }
              }
            >
              <i
                aria-hidden="true"
                className="green check circle outline fitted icon"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              />
            </span>
          </h2>
        </div>
        <div
          className="description"
        >
          <div
            className="content"
          >
            Number of priests
            : 
            12
            <br />
            Town
            : 
            Test city
          </div>
        </div>
      </div>
    </a>
  </div>,
  <div
    className="ui center aligned container"
    style={
      Object {
        "marginTop": "3em",
      }
    }
  >
    <div
      aria-label="Pagination Navigation"
      className="ui pagination menu"
      role="navigation"
    >
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="First item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="firstItem"
        value={1}
      >
        «
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Previous item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="prevItem"
        value={1}
      >
        ⟨
      </a>
      <a
        aria-current={true}
        aria-disabled={false}
        className="active item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={1}
      >
        1
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Next item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="nextItem"
        value={1}
      >
        ⟩
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Last item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="lastItem"
        value={1}
      >
        »
      </a>
    </div>
  </div>,
]
`;

exports[`SearchParishesContainer element should render a placeholder card and a message when the parishes array is empty 1`] = `
Array [
  <div
    className="ui fluid card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="header"
          >
            <div
              className="line"
            />
          </div>
        </div>
      </div>
      <div
        className="description"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
        </div>
      </div>
    </div>
  </div>,
  <h2
    className="ui header"
  >
    No results were found for your search. Try widening the search criterias.
  </h2>,
]
`;

exports[`SearchParishesContainer element should render a placeholder card when only loading is true 1`] = `
<div
  className="ui fluid card"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="header"
        >
          <div
            className="line"
          />
        </div>
      </div>
    </div>
    <div
      className="description"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchParishesContainer element should render a placeholder card when only loading is true, for map view 1`] = `
<div
  className="ui fluid card"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="header"
        >
          <div
            className="line"
          />
        </div>
      </div>
    </div>
    <div
      className="description"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchParishesContainer element should render an error message when showError and loading are both true 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "7em",
    }
  }
>
  <h2
    className="ui header red"
  >
    Could not estabilish server connection.
  </h2>
  <h3
    className="ui header"
  >
    <a
      aria-current="page"
      className="hoverLink active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Back to the Home Page
    </a>
  </h3>
</div>
`;

exports[`SearchParishesContainer element should render an error message when showError and loading are both true, for map view 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "7em",
    }
  }
>
  <h2
    className="ui header red"
  >
    Could not estabilish server connection.
  </h2>
  <h3
    className="ui header"
  >
    <a
      aria-current="page"
      className="hoverLink active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Back to the Home Page
    </a>
  </h3>
</div>
`;

exports[`SearchParishesContainer element should render an error message when showError is true 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "7em",
    }
  }
>
  <h2
    className="ui header red"
  >
    Could not estabilish server connection.
  </h2>
  <h3
    className="ui header"
  >
    <a
      aria-current="page"
      className="hoverLink active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Back to the Home Page
    </a>
  </h3>
</div>
`;

exports[`SearchParishesContainer element should render an error message when showError is true, for map view 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "7em",
    }
  }
>
  <h2
    className="ui header red"
  >
    Could not estabilish server connection.
  </h2>
  <h3
    className="ui header"
  >
    <a
      aria-current="page"
      className="hoverLink active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Back to the Home Page
    </a>
  </h3>
</div>
`;
